server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -422,6 +451,59 @@ public static void ensureNodeCommissioned(DiscoveryNode node, Metadata metadata)
         }
     }
 
+    /**
+     * The method ensures two conditions -
+     * 1. The joining node is remote store if it is joining a remote store cluster.
```

Comment:
nit :`The joining node is remote store  backed` instead .

Reply:
Also lets sumarize it two points 

1. ensure homogenity of all nodes - either all are remote backed or none are remote backed
2. Validates that all repositories are exact same for all nodes if they are remote backed .